modules/kubeconfig/main.tf

```diff
@@ -38,8 +39,25 @@ data "template_file" "kubeconfig_sa" {
     token        = lookup(data.kubernetes_secret.sa_secret.0.data,"token", "")
     namespace    = var.namespace
   }
+  depends_on = [data.kubernetes_secret.sa_secret]
 }
 
+# 1.24 change: Create service account secret
+resource "kubernetes_secret" "sa_secret" {
+  count = var.create_static_kubeconfig ? 1 : 0
+  metadata {
+    name      = local.service_account_secret_name
+    namespace = var.namespace
+    annotations = {
+      "kubernetes.io/service-account.name" = local.service_account_name
+    }
+  }
+  type = "kubernetes.io/service-account-token"
+  depends_on = [kubernetes_service_account.kubernetes_sa]
```

Comment:
Guessing this is needed for its assignment and that if the service account is not there this will fail correct?

Reply:
That's correct, need to have SA that exist before we can create a secret associated with it.